### PR TITLE
Adds the Compliance Feature on the Child Register Page

### PR DESCRIPTION
**Pull request by paulinembabu**
See issue: https://github.com/OpenSRP/opensrp-client-emmunize-malawi/issues/9

### CODE REVIEW

File: opensrp-child/src/main/java/org/smartregister/child/task/VaccinationAsyncTask.java [hunk 1]
```diff
@@ -135,6 +136,14 @@ private void executeInBackground() {
             }
         }
 
+        overDueCount = 0;
+        for (Alert alert : alerts) {
+            String name = alert.visitCode().trim().replace(" ", "").toLowerCase();
```
Comment (bennsimon) [inline opensrp-child/src/main/java/org/smartregister/child/task/VaccinationAsyncTask.java:141]: Check for null here on ` alert.visitCode()`

File: opensrp-child/src/main/java/org/smartregister/child/task/VaccinationAsyncTask.java [hunk 2]
```diff
@@ -185,16 +196,34 @@ private void updateRecordVaccination(VaccineViewRecordUpdateWrapper updateWrappe
...
         Map<String, Object> nv = updateWrapper.getNv();
         DateTime dueDate = null;
-        if (nv != null) {
+
+        complianceCheck.setVisibility(View.VISIBLE);
+        complianceText.setText(context.getString(R.string.number_missed, overDueCount));
+        if (overDueCount == 0) {
+            complianceCheck.setImageResource(R.drawable.ic_action_check);
+        } else if (overDueCount == 1) {
+            complianceCheck.setImageResource(R.drawable.ic_yelllow_flag);
```
Comment (bennsimon) [inline opensrp-child/src/main/java/org/smartregister/child/task/VaccinationAsyncTask.java:220]: typo here `ic_yelllow_flag`

File: opensrp-child/src/main/res/layout/child_register_header.xml [hunk 3]
```diff
@@ -7,7 +7,7 @@
     android:background="@color/client_list_header_dark_grey"
     android:orientation="horizontal"
 
-    android:weightSum="10">
+    android:weightSum="13.4">
```
Comment (bennsimon) [inline opensrp-child/src/main/res/layout/child_register_header.xml:10]: Will this design change affect other projects?

**Review by bennsimon — CHANGES_REQUESTED** [review]

File: opensrp-child/src/main/java/org/smartregister/child/task/VaccinationAsyncTask.java (hunk 1 above)
Comment (paulinembabu) [reply opensrp-child/src/main/java/org/smartregister/child/task/VaccinationAsyncTask.java]: done

File: opensrp-child/src/main/java/org/smartregister/child/task/VaccinationAsyncTask.java (hunk 2 above)
Comment (paulinembabu) [reply opensrp-child/src/main/java/org/smartregister/child/task/VaccinationAsyncTask.java]: Edited

File: opensrp-child/src/main/res/layout/child_register_header.xml (hunk 3 above)
Comment (paulinembabu) [reply opensrp-child/src/main/res/layout/child_register_header.xml]: Have Put the WeightSum to 10.6 and made sure that it's not affecting other Implementations

File: opensrp-child/src/main/java/org/smartregister/child/provider/ChildRegisterProvider.java [hunk 4]
```diff
@@ -146,6 +144,12 @@ public RegisterViewHolder createViewHolder(ViewGroup parent) {
         view.findViewById(R.id.child_next_appointment_wrapper).setVisibility(ChildLibrary.getInstance().getProperties()
                 .getPropertyBoolean(Constants.PROPERTY.HOME_NEXT_VISIT_DATE_ENABLED) ? View.VISIBLE : View.GONE);
 
+        if (ChildLibrary.getInstance().getProperties().hasProperty(Constants.PROPERTY.HOME_COMPLIANCE_ENABLED)) {
```
Comment (bennsimon) [inline opensrp-child/src/main/java/org/smartregister/child/provider/ChildRegisterProvider.java:147]: Is this necessary isnt the default from `ChildLibrary.getInstance().getProperties().getPropertyBoolean` `false`

File: opensrp-child/src/main/java/org/smartregister/child/provider/ChildRegisterProvider.java [hunk 5]
```diff
@@ -228,6 +232,13 @@ private void populatePatientColumn(CommonPersonObjectClient pc, SmartRegisterCli
         recordVaccination.setTag(R.id.record_action, Constants.RECORD_ACTION.VACCINATION);
 
 
+        View showCompliance = viewHolder.showCompliance;
+        showCompliance.setTag(client);
+        showCompliance.setOnClickListener(onClickListener);
+        if (ChildLibrary.getInstance().getProperties().hasProperty(ChildAppProperties.KEY.HOME_COMPLIANCE_ENABLED)) {
```
Comment (bennsimon) [inline opensrp-child/src/main/java/org/smartregister/child/provider/ChildRegisterProvider.java:238]: Is this necessary isnt the default from `ChildLibrary.getInstance().getProperties().getPropertyBoolean` `false`

File: opensrp-child/src/main/java/org/smartregister/child/task/VaccinationAsyncTask.java [hunk 6]
```diff
@@ -135,6 +136,16 @@ private void executeInBackground() {
             }
         }
 
+        overDueCount = 0;
+        if(alerts != null){
+            for (Alert alert : alerts) {
+                String name = alert.visitCode().trim().replace(" ", "").toLowerCase();
```
Comment (bennsimon) [inline opensrp-child/src/main/java/org/smartregister/child/task/VaccinationAsyncTask.java:142]: Check for null on ` alert.visitCode()`

File: opensrp-child/src/main/res/layout/child_register_header.xml [hunk 7]
```diff
@@ -7,7 +7,7 @@
     android:background="@color/client_list_header_dark_grey"
     android:orientation="horizontal"
 
-    android:weightSum="10">
+    android:weightSum="10.6">
```
Comment (bennsimon) [inline opensrp-child/src/main/res/layout/child_register_header.xml:10]: Does this change affect other projects?

File: opensrp-child/src/main/java/org/smartregister/child/util/ChildAppProperties.java [hunk 8]
```diff
@@ -21,6 +21,7 @@
         public static final String HOME_RECORD_WEIGHT_ENABLED = "home.record.weight.enabled";
         public static final String HOME_TOOLBAR_SCAN_CARD_ENABLED = "home.toolbar.scan.card.enabled";
         public static final String HOME_TOOLBAR_SCAN_QR_ENABLED = "home.toolbar.scan.qr.enabled";
+        public  static final String HOME_COMPLIANCE_ENABLED = "home.compliance.enabled";
```
Comment (bennsimon) [inline opensrp-child/src/main/java/org/smartregister/child/util/ChildAppProperties.java:24]: kindly add some description in the ReadMe

**Review by bennsimon — CHANGES_REQUESTED** [review]

File: opensrp-child/src/main/res/layout/child_register_header.xml (hunk 7 above)
Comment (paulinembabu) [reply opensrp-child/src/main/res/layout/child_register_header.xml:10]: No it doesn't

File: opensrp-child/src/main/java/org/smartregister/child/provider/ChildRegisterProvider.java (hunk 4 above)
Comment (paulinembabu) [reply opensrp-child/src/main/java/org/smartregister/child/provider/ChildRegisterProvider.java:147]: Yes, It is. This is the Check that is made to Show or Hide on the App. So yes the default is False but it changes on the app

File: opensrp-child/src/main/java/org/smartregister/child/provider/ChildRegisterProvider.java (hunk 5 above)
Comment (paulinembabu) [reply opensrp-child/src/main/java/org/smartregister/child/provider/ChildRegisterProvider.java:238]: Yes, It is. This is the Check that is made to Show or Hide on the App. So yes the default is False but it changes on the app

**Review by bennsimon — APPROVED** [review]

**Pull request merged**